src/vellum/workflows/nodes/bases/tests/test_string_comparison_error.py

```diff
@@ -0,0 +1,60 @@
+"""
+Test for APO-1936: Conditional nodes internal server error when doing string greater than or equal comparisons.
+
+This test reproduces the error at the node level by creating a workflow node with ports that use
+comparison expressions on string values.
+"""
+
+import pytest
+
+from vellum.workflows.descriptors.exceptions import InvalidExpressionException
+from vellum.workflows.inputs.base import BaseInputs
+from vellum.workflows.nodes.bases.base import BaseNode
+from vellum.workflows.outputs.base import BaseOutputs
+from vellum.workflows.ports.port import Port
+from vellum.workflows.state.base import BaseState, StateMeta
+
+
+def test_conditional_node__string_greater_than_or_equal_comparison_error():
+    """
+    Test that reproduces the internal server error when doing string greater than or equal comparisons.
+    This should raise a user-facing error instead of allowing string comparisons.
+
+    Reproduces: https://linear.app/vellum/issue/APO-1936
+    """
+
+    # GIVEN workflow inputs with a string field and a float value
+    class Inputs(BaseInputs):
+        field: str
+        value: float
+
+    class State(BaseState):
+        pass
+
+    # AND a conditional node that compares a string to a float with >=
+    class ConditionalNode(BaseNode[State]):
+        class Ports(BaseNode.Ports):
+            if_port = Port.on_if(Inputs.field.greater_than_or_equal_to(Inputs.value))
+            else_port = Port.on_else()
+
+        class Outputs(BaseOutputs):
+            result: str
+
+        def run(self) -> Outputs:
+            return self.Outputs(result="executed")
+
+    # WHEN we initialize the node with a string field and float value
+    state = State(
+        meta=StateMeta(
+            workflow_inputs=Inputs(field="hello", value=42.0),
+        )
+    )
+
+    node = ConditionalNode(state=state)
+
+    # THEN it should raise InvalidExpressionException when evaluating the port condition
+    with pytest.raises(InvalidExpressionException) as exc_info:
+        node.Ports.if_port.resolve_condition(state)
```

Comment:
Ah interesting. Ok expand this test to use within the context of a workflow, and add as a `tests/workflows` module